addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -69,6 +75,8 @@ odoo.define('website_slides.quiz', function (require) {
                 self._renderAnswers();
                 self._renderAnswersHighlighting();
                 self._renderValidationInfo();
+                self.courseJoinWidget = new CourseJoinWidget(self);
+                self.$('.o_wslides_js_course_join_link').click(self.courseJoinWidget._onClickJoin.bind(self));
```

Comment:
Can't the `CourseJoinWidget` be autonomous, and catch the event itself ?
I'm sure the JS team wouldn't approve this. 
Can't built the widget (it should have a template) and `appendTo` somewhere on the parent widget ?

Reply:
If it can't be autonomous then it should throw a custom event like 'on_course_join' and this widget should catch that instead.

Plus indeed it should be something like `new CourseJoinWidget(self).attachTo(self.$('o_course_join_widget'))`

Reply:
Done